cookiecutter/{{ cookiecutter.package_name }}/e2e/e2e_utils.py

```diff
@@ -0,0 +1,140 @@
+import contextlib
+import logging
+import os
+import shlex
+import socket
+import subprocess
+import sys
+import time
+import typing
+from contextlib import closing
+from tempfile import TemporaryFile
+
+import requests
+
+
+LOGGER = logging.getLogger(__file__)
+
+
+def _find_free_port():
+    with closing(socket.socket(socket.AF_INET, socket.SOCK_STREAM)) as s:
```

Comment:
It would be great if the code in our templates had more docstrings. For each method you can write something to make the code easier to use. We should also type hints.

Reply:
If it's ok I will do this in all utils files sync PR